index.js

```diff
@@ -1,55 +1,61 @@
-'use strict';
-
 const AWS = require('aws-sdk');
+const Sharp = require('sharp');
+
 const S3 = new AWS.S3({
   signatureVersion: 'v4',
 });
-const Sharp = require('sharp');
 
 const BUCKET = process.env.BUCKET;
 const URL = process.env.URL;
 const ALLOWED_DIMENSIONS = new Set();
 
 if (process.env.ALLOWED_DIMENSIONS) {
   const dimensions = process.env.ALLOWED_DIMENSIONS.split(/\s*,\s*/);
-  dimensions.forEach((dimension) => ALLOWED_DIMENSIONS.add(dimension));
+  dimensions.forEach(dimension => ALLOWED_DIMENSIONS.add(dimension));
 }
 
-exports.handler = function(event, context, callback) {
+exports.handler = function handler(event, context, callback) {
   const key = event.queryStringParameters.key;
-  const match = key.match(/((\d+)x(\d+))\/(.*)/);
+  const match = key.match(/((\d+|auto)x(\d+|auto))\/(.*)/);
   const dimensions = match[1];
-  const width = parseInt(match[2], 10);
-  const height = parseInt(match[3], 10);
+  const width = match[2] === 'auto' ? null : parseInt(match[2], 10);
+  const height = match[3] === 'auto' ? null : parseInt(match[3], 10);
   const originalKey = match[4];
 
-  if(ALLOWED_DIMENSIONS.size > 0 && !ALLOWED_DIMENSIONS.has(dimensions)) {
-     callback(null, {
+  if (ALLOWED_DIMENSIONS.size > 0 && !ALLOWED_DIMENSIONS.has(dimensions)) {
+    callback(null, {
```

Comment:
It's not part of this PR, but just curious why this status is a 403

Reply:
403 is Forbidden, So I think the idea is that the provided size is not allowed, and is thus forbidden, but we've pretty consistently used 400s in the case of bad parameters, so I'm fine to change it to a 400.